### ISSUE TITLE
[no-unused-vars] All enum entries are seen unused

### ISSUE THREAD

**Issue by tpetillot** [typescript-eslint/typescript-eslint#2870]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
}
```

```TS
export enum Enum {
  A,
  B,
  C
}

// Any usage of the enum
// if in another file, import {Enum} from "./enum"; but issue also triggered in same file

console.log(Enum.A);
console.log(Enum.B);
console.log(Enum.C);
```

**Expected Result**

No errors

**Actual Result**

```
  2:3  error  'A' is defined but never used  no-unused-vars
  3:3  error  'B' is defined but never used  no-unused-vars
  4:3  error  'C' is defined but never used  no-unused-vars
```

**Additional Info**

Command used `eslint --quiet 'src/**/*.ts'`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.1` |
| `@typescript-eslint/parser`        | `4.9.1` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.15.1` |
| `node`                             | `14.14.0` |

**Comment by mattbrictson** [comment]
@tpetillot I am seeing the same problem after upgrading from 4.8.0 to 4.10.0. I see that you closed the issue, though. What is the fix?

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

Also [issue search](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+sort%3Aupdated-desc+in%3Atitle+%5Bno-unused-vars%5D+enum+is%3Aclosed+) reveals previous issues

**Comment by mattbrictson** [comment]
Never mind, I was mistakenly using `no-unused-vars` instead of `@typescript-eslint/no-unused-vars`. 😅
